### ISSUE TITLE
Self-Signed CA

### ISSUE THREAD

**Issue by sebgoa** [bitnami/cabin#12]
_From @Bregor on September 6, 2016 18:13_

At first, thank you for the wonderful app!

It will be perfect if you add a possibility to use self-signed CAs which will grant access to test and development clusters.

Thanks in advance.

_Copied from original issue: skippbox/cabin-issues#13_

**Comment by sebgoa** [comment]
Hi, right now the TLS verification is actually disabled.

What problem are you having exactly ?

**Comment by sebgoa** [comment]
_From @Bregor on September 6, 2016 19:39_

@runseb I have some k8s-clusters for QA needs which are world accessible.
For the authentication, I use tokens and public part of CA.

Clusters looks like following:

```
clusters:
- cluster:
    certificate-authority-data: <ca.pem here>
    server: https://<qa1>:8443
  name: qa1
```

The main problem is with a certificate-authority part. There is no place in **Cabin** to put it in and without CA my clusters reject authentications.

**Comment by sebgoa** [comment]
will you be able to copy/paster the ca.pem from somewhere else on your device into Cabin ?

**Comment by sebgoa** [comment]
_From @Bregor on September 7, 2016 17:2_

Of course.

**Comment by sebgoa** [comment]
We still have this opened. it is not forgotten

**Comment by sebgoa** [comment]
@Bregor so you are not using a client certificate and client key for authentication. It is just that you want the connection to trust your self signed CA used for the cert of the server ?

**Comment by sebgoa** [comment]
_From @Bregor on November 28, 2016 14:41_

That's it for now.
In the beginning, we were using cert-based auth, but long ago we switched to bearer tokens.
But our certs are still self-signed for now.

**Comment by sebgoa** [comment]
ok, so I am still struggling to understand why if we disable TLS verification (for self signed CA) you would still need to pass the root CA cert. Since the _cluster_ CA is not used to sign the client requests in any way...

Meaning, if you use a self-signed CA now for your API server, I am pretty sure Cabin still works. Once we add the possibility to specify the CA, this will allow us to check the TLS connection (which of course is so much better). But I don't think the connection to the cluster is prevented server side.

**Comment by sebgoa** [comment]
_From @Bregor on November 28, 2016 14:57_

To be quite fair, now I tried the last update with my fake-certed cluster, and it works!
It seems we can close this issue.
I'll try with some more test clusters if you don't mind.

**Comment by sebgoa** [comment]
ok, we do have a solution for cert-based auth in the next version....and we are looking into enabling TLS verify for self-signed CA.

**Comment by sebgoa** [comment]
_From @ipedrazas on January 11, 2017 11:40_

any update on this?

**Comment by sebgoa** [comment]
@ipedrazas it should be in the latest version.

**Comment by strigazi** [comment]
I'm using version 0.3.1 , will there be a new release to support self signed CA?

**Comment by kemcake** [comment]
@strigazi there's an issue on Android in 0.3.1 with self-signed certificate, I'm working on it 🙌

**Comment by kemcake** [comment]
Should be fixed in 0.4.X thanks to #72
